operations/configuration_management/Enable_Ansible_Profiling.md

```diff
@@ -1,26 +1,67 @@
 
-1.  Edit the `cfs-default-ansible-cfg` ConfigMap.
+## Enabling execution time profiling
 
-    ```bash
-    kubectl edit cm cfs-default-ansible-cfg -n services
-    ```
+1. Edit the `cfs-default-ansible-cfg` ConfigMap.
 
-1.  Uncomment the indicated line by removing the \# character from the beginning of the line.
+   ```bash
+   kubectl edit cm cfs-default-ansible-cfg -n services
+   ```
 
-    ```yaml
-    #callback_whitelist    = cfs_aggregator, timer, profile_tasks, profile_roles
-    ```
+1. Uncomment the indicated line by removing the \# character from the beginning of the line.
 
-1.  Comment out the indicated line by adding a \# character to the beginning of the line.
+   ```yaml
+   #callback_whitelist    = cfs_aggregator, timer, profile_tasks, profile_roles
+   ```
 
-    ```yaml
-    callback_whitelist    = cfs_aggregator
-    ```
+1. Comment out the indicated line by adding a \# character to the beginning of the line.
 
-New sessions will be created with profiling information available in the Ansible logs of the session pods. Alternatively, if editing the default `ansible.cfg` file that CFS uses, is not desired, a new Ansible configuration can also be created to enable profiling and direct CFS to use it. See [Use a Custom `ansible.cfg` File](Use_a_Custom_ansible-cfg_File.md) for more information.
+   ```yaml
+   callback_whitelist    = cfs_aggregator
+   ```
 
+New sessions will be created with profiling information available in the Ansible logs of the session pods.
+
+## Enabling memory profiling
+
+Ansible memory profiling is somewhat limited by the constraints of running Ansible inside Kubernetes containers.
+The profiling is useful for determining the relative memory consumption of different tasks,
+but the exact values will include some slight overhead from the container due to the fact that a separate `cgroup` can not be easily created in the containers read-only file system.
+Similarly, the play's `Execution Maximum` will incorrectly show a value of 0 since the workaround for Ansible's attempts to write to a read-only file is to direct it to a temporary file.
+
+1. Edit the `cfs-default-ansible-cfg` ConfigMap.
+
+   ```bash
+   kubectl edit cm cfs-default-ansible-cfg -n services
+   ```
+
+1. Add `cgroup_memory_recap` to the `callback_whitelist` line that is currently uncommented.
+This can be added to the execution time profiling tasks if desired, or it can be added on its own.  
+
+   ```yaml
+   callback_whitelist    = cfs_aggregator, cgroup_memory_recap
+   ```
+
+   or
+
+   ```yaml
+   callback_whitelist    = cfs_aggregator, timer, profile_tasks, profile_roles, cgroup_memory_recap
+   ```
+
+1. Add the following section to the configuration.
+Note that the value for `max_mem_file` can not point to `/sys/fs/cgroup/memory` as that is read-only in the container and Ansible will try to write to the `memory.max_usage_in_bytes` file.
```

Comment:
```suggestion
1. Add the following section to the configuration.

   > Note that the value for `max_mem_file` can not point to `/sys/fs/cgroup/memory`, because that is read-only in the container and Ansible will try to write to the `memory.max_usage_in_bytes` file.
```